CMakeLists.txt

```diff
@@ -76,6 +81,19 @@ function(pk_patch_file old_string new_string path)
     file(WRITE ${path} "${shader_content}")
 endfunction()
 
+if(DEFINED O3DE_VERSION_MAJOR)
+    # Between engine version 1.2.0 -> 2.2.0 there is an issue that prevents
+    # PopcornFX from being added to the .setreg files for loading .dlls at runtime
+    # In addition there is a bug where the engine version in the SDK is the display version
+    # instead of matching the version field in the source code
+    if((O3DE_VERSION_MAJOR EQUAL 23 AND O3DE_VERSION_MINOR EQUAL 05 AND O3DE_VERSION_PATCH EQUAL 0) OR 
+       (O3DE_VERSION_MAJOR EQUAL 1 AND O3DE_VERSION_MINOR GREATER_EQUAL 2) OR 
+       (O3DE_VERSION_MAJOR EQUAL 2 AND O3DE_VERSION_MINOR LESS_EQUAL 1))
+        message(VERBOSE "Activating PopcornFX gem for O3DE ${PK_O3DE_MAJOR_VERSION}")
+        ly_enable_gems(GEMS PopcornFX)
```

Comment:
This does work as a workaround.

Since the PopcornFX gem is not normally registered in the  "external_subdirectories" field of a project's `project.json` file or the O3DE engine `engine.json` file, the only time it will get added to a project is when it is already active at this point, so explicitly calling `ly_enable_gems` adds the PopcornFX gem to the `LY_DELAYED_ENABLE_GEMS__NO_PROJECT__` global property which gets iterated at the end of the CMake configure step in [ly_enable_gems_delayed](https://github.com/o3de/o3de/blob/development/cmake/Gems.cmake#L413) to add a TARGET dependency to the O3DE application targets which have a Gem variant set on it via a call to `ly_set_gem_variant_to_load` such as the Editor, AssetProcessor and GameLauncher.